### PR TITLE
Tokenutilize Azure auth

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: path_role.go [hunk 1]
```diff
@@ -10,12 +10,13 @@ import (
...
 	"github.com/hashicorp/vault/sdk/helper/policyutil"
+	"github.com/hashicorp/vault/sdk/helper/tokenutil"
 	"github.com/hashicorp/vault/sdk/logical"
 )
 
 // pathsRole returns the path configurations for the CRUD operations on roles
 func pathsRole(b *azureAuthBackend) []*framework.Path {
-	return []*framework.Path{
+	p := []*framework.Path{
 		&framework.Path{
```
Comment (kalafut) [inline path_role.go]: While we're here, consider removing the reduntant `&framework.Path`s and `&framework.FieldSchema`s.

File: path_role.go [hunk 2]
```diff
@@ -243,39 +273,99 @@ func (b *azureAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logica
...
-	if policiesRaw, ok := data.GetOk("policies"); ok {
-		role.Policies = policyutil.ParsePolicies(policiesRaw)
-	}
+	if err := role.ParseTokenFields(req, data); err != nil {
+		return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
+	}
+
+	// Handle upgrade cases
+	{
+		policiesRaw, ok := data.GetOk("token_policies")
```
Comment (kalafut) [inline path_role.go]: In all of these cases, the result term (e.g. `policiesRaw` here) isn't used, but the variable name _is_ used in inner blocks. Probably clearer to use `_` for the top level `GetOK`.

File: path_role.go [hunk 3]
```diff
@@ -243,39 +273,99 @@ func (b *azureAuthBackend) pathRoleCreateUpdate(ctx context.Context, req *logica
...
+			}
+		}
 	}
 
-	if tokenTTLRaw, ok := data.GetOk("ttl"); ok {
-		role.TTL = time.Duration(tokenTTLRaw.(int)) * time.Second
-	} else if req.Operation == logical.CreateOperation {
-		role.TTL = time.Duration(data.Get("ttl").(int)) * time.Second
+	if role.TokenPeriod > b.System().MaxLeaseTTL() {
+		return logical.ErrorResponse(fmt.Sprintf("token period of '%q' is greater than the backend's maximum lease TTL of '%q'", role.TokenPeriod.String(), b.System().MaxLeaseTTL().String())), nil
```
Comment (kalafut) [inline path_role.go]:
```suggestion
		return logical.ErrorResponse("token period of %q is greater than the backend's maximum lease TTL of %q", role.TokenPeriod.String(), b.System().MaxLeaseTTL().String()), nil
```

Comment (jefferai) [reply path_role.go]: ErrorResponse takes in a bare string...

Comment (kalafut) [reply path_role.go]: It can also take a format string plus  parameters. And %q already adds wrapping quotes, so the extras were removed.

Comment (jefferai) [reply path_role.go]: ...since when? :-)

Comment (jefferai) [reply path_role.go]: I don't really know why it had quotes plus %q and didn't really look, I just kept the error as is and just adjusted the parameter

**Pull request merged**